plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcConnectorTest.java

```diff
@@ -327,4 +352,57 @@ protected Session joinPushdownEnabled(Session session)
                 .setCatalogSessionProperty(session.getCatalog().orElseThrow(), "join_pushdown_enabled", "true")
                 .build();
     }
+
+    @Test(timeOut = 60_000)
+    public void testCancellation()
+            throws Throwable
+    {
+        if (!hasBehavior(SUPPORTS_CANCELLATION)) {
+            throw new SkipException("Cancellation is not supported by given connector");
+        }
+
+        Duration minimalQueryDuration = new Duration(1, MINUTES);
+
+        // verify that long lasting query is correct
+        assertQuerySucceeds("EXPLAIN " + getLongLastingQuery(minimalQueryDuration));
+
+        List<RemoteDatabaseEvent> remoteEvents = captureRemoteEventsDuring(() -> {
+            String longLastingQuery = getLongLastingQuery(minimalQueryDuration);
+            Future<?> future = threadPool.submit(() -> assertQueryFails(longLastingQuery, "Query killed. Message: Killed by test"));
+            QueryId queryId = getQueryId(longLastingQuery);
+            sleep(1_000); // to make sure remote data source is queried
+            assertUpdate(format("CALL system.runtime.kill_query(query_id => '%s', message => '%s')", queryId, "Killed by test"));
+            sleep(1_000); // to make sure query is cancelled
+            future.get();
+        });
+
+        assertThat(remoteEvents).isNotEmpty();
+        assertThat(remoteEvents.stream())
+                .map(RemoteDatabaseEvent::getStatus)
+                .contains(CANCELLED);
+    }
+
+    private QueryId getQueryId(String sqlPattern)
+    {
+        AtomicReference<QueryId> queryId = new AtomicReference<>();
+        assertEventually(() -> {
+            MaterializedResult queriesResult = getQueryRunner().execute(format(
+                    "SELECT query_id FROM system.runtime.queries WHERE state = 'RUNNING' AND query LIKE '%%%s%%' AND query NOT LIKE '%%system.runtime.queries%%'",
+                    sqlPattern));
+            assertEquals(queriesResult.getRowCount(), 1);
+            queryId.set(new QueryId((String) queriesResult.getOnlyValue()));
+        });
+        return queryId.get();
+    }
+
+    protected List<RemoteDatabaseEvent> captureRemoteEventsDuring(CheckedRunnable runnable)
```

Comment:
> it can be hard to implement this interface for other databases.

I already implemented this for other database. Basically it is `SELECT * FROM  system.runtime.queries`. I will try to simplify this.